[S Dinesh Raj] ip

Dineshraj555 opened the pull request:
# Mike

> “Your mind is for having ideas, not holding them.” – David Allen ([Source](https://dansilvestre.com/productivity-quotes/))

Mike frees your mind of having to remember things yo need to do. It's,
* text-based
* easy to learn
* ~~FAST~~*SUPER* FAST to use

All you need to do is, 
1. download it from [here](https://nus-cs2103-ay2122s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :exploding_head:

And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

##

If you Java programmer, you can use it to practice Java too. Here's the ```main``` method: 

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Dineshraj555 committed:
Level-1: Greet, Echo, Exit

Dineshraj555 committed:
Level-2: Add, List

Dineshraj555 committed:
Level-3

Dineshraj555 committed:
Level-4

Dineshraj555 committed:
A-TextUiTesting: Automated Text UI Testing

Dineshraj555 committed:
Minor changes due to indentation

Dineshraj555 committed:
A-Exceptions

Dineshraj555 committed:
Level-6: Delete

Dineshraj555 committed:
Level 7: Save function implementation

Dineshraj555 committed:
Level 8. Addition of Dates and Times

Dineshraj555 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Level 8. Addition of Dates and Times

Dineshraj555 committed:
A-MoreOOP: Added UI, Storage, Tasklist and Parser classes to improve OOP

Dineshraj555 committed:
A-Packages: Organised the various classes into Packages

Dineshraj555 committed:
Minor changes to output of Task: toSave() method

Dineshraj555 committed:
A-JUnit: Added JUnit Tests for Event and Deadline Test

Dineshraj555 committed:
Made minor changes to code in terms of organisation after review.

Dineshraj555 committed:
A-JavaDoc: Addition of Java documentation to the code base.

Dineshraj555 committed:
A-CodingStandard: Made changes to code to align with Module coding standard

Dineshraj555 committed:
Level 9: Added find function

Dineshraj555 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  A-CodingStandard: Made changes to code to align with Module coding standard

# Conflicts:
#	data/duke.txt
#	src/main/java/Task/Event.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/UI.java
#	src/main/java/task/TaskList.java

Dineshraj555 committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Level 9: Added find function

# Conflicts:
#	data/duke.txt
#	src/main/java/Task/Task.java
#	src/main/java/Task/TaskList.java

Dineshraj555 committed:
Merge branch 'add-gradle-support' of https://github.com/Dineshraj555/ip into A-Gradle

* 'add-gradle-support' of https://github.com/Dineshraj555/ip:
  Add Gradle support

Dineshraj555 committed:
A-Gradle: Add Gradle to Duke

yongler commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,84 @@
+    public void processData(TaskList tasklist) throws IOException {
+
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        String cmd = br.readLine();
```
Maybe a more descriptive name can be used? e.g. command. I noticed this on some other parts of the code too

yongler commented on src/main/java/duke/UI.java:
```diff
@@ -0,0 +1,118 @@
+        String start = "              Greetings! Mike here!\n" +
+                "               How can I help you?\n" ;
+        String logo =
+                "||======||==||======|| !!!! ||====||    //===//======||\n"
```
Maybe line length should be no longer than 120 chars for readability?

yongler commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+    /**
+     * List to house the Tasks.
+     */
+    private ArrayList<Task> list;
```
Maybe plural form should be used on names representing a collection of objects? e.g. tasks

joshuayapwj98 replied:
A suggestion, maybe a more descriptive term could be used for the list? like tasks

yongler commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+        UI.printLine();
+    }
+
+    public void find(String str) {
```
Maybe a more descriptive method name could be used?

yongler reviewed (commented):
LGTM, just some minor nits to fix!

joshuayapwj98 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,83 @@
+            BufferedReader r = new BufferedReader(new FileReader(file));
+            String t = r.readLine();
+
+            while (t != null) {
+                char i = t.charAt(0);
+                if (i == 'T') {
+                    String[] in = t.split(":");
+                    list.add(new Todo(in[2]));
+                } else if (i == 'D') {
+                    String[] in = t.split(":");
+                    list.add(new Deadline(in[2], in[3]));
+                } else if (i == 'E') {
+                    String[] in = t.split(":");
+                    list.add(new Event(in[2], in[3]));
+                } else {
+                    // do nothing
+                }
+                t = r.readLine();
+            }
+        }
```
A neat set of code!

However, I was wondering if you could reformat this? It follows an arrow style where code is indented outward.

Nonetheless, Good job!! Very intuitive!

joshuayapwj98 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+     */
+    public void printTaskList() {
+        UI.printLine();
+        for(int m = 0; m < list.size(); m++) {
```
Intuitive method! Just a suggestion:

```suggestion
for (int m = 0; m < list.size(); m++) {
```

joshuayapwj98 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+                output.add(list.get(m));
+            } else if (scan(in, str)){
+                output.add(list.get(m));
+            } else {
+                // add nothing
+            }
```
Perhaps the if-else could be reformatted so that there isn't any unused code section?

joshuayapwj98 commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,135 @@
+
+    private void printSearchList(ArrayList<Task> list) {
+        UI.printLine();
+        for(int m = 0; m < list.size(); m++) {
```
Another suggestion:
```suggestion
for (int m = 0; m < list.size(); m++) {
```

joshuayapwj98 reviewed (commented):
LGTM, Overall, elegant set of code!

Just minor indentation issues but good work! 👍🏼

Dineshraj555 committed:
Working GUI: Need to fine tune the code

Dineshraj555 committed:
updates to GUI

Dineshraj555 committed:
Merge branch 'Level-10'

* Level-10:
  updates to GUI
  Working GUI: Need to fine tune the code

Dineshraj555 committed:
Made basic formatting changes to code base.

Dineshraj555 committed:
Add assertions to code

The code base did not include any assertions up till now to document
important assumptions that should hold at various points in the code.

Add Assertions to the code base wherever it deemed necessary.

Dineshraj555 committed:
Improve Code quality

The code contains irregularities which are not align with the module
coding standard and quality.

Improve code quality by vetting through the code base and change points
where violations of quality has been observed.

Code has been refactor following KISS and SLAP principles.

Dineshraj555 committed:
Merge pull request #1 from Dineshraj555/branch-A-Assertions

Add assertions to code

Dineshraj555 committed:
Merge branch 'master' of https://github.com/Dineshraj555/ip

* 'master' of https://github.com/Dineshraj555/ip:
  Add assertions to code

Dineshraj555 committed:
Merge branch 'master' into A-CodeQuality

Dineshraj555 committed:
Merge pull request #2 from Dineshraj555/A-CodeQuality

Improve Code quality

Dineshraj555 committed:
Merge branch 'master' of https://github.com/Dineshraj555/ip

* 'master' of https://github.com/Dineshraj555/ip:
  Improve Code quality

Dineshraj555 committed:
Minor Patch up to Parser and Storage.

Dineshraj555 committed:
Add tag features as part of BCD Extension.

Dineshraj555 committed:
Merge branch 'BCD-Extension'

* BCD-Extension:
  Add tag features as part of BCD Extension.

Dineshraj555 committed:
Better GUI, Basic improvements

Dineshraj555 committed:
Update README.md

Dineshraj555 committed:
Update README.md

Dineshraj555 committed:
Update README.md

Dineshraj555 committed:
Update README.md

Dineshraj555 committed:
Update README.md

Dineshraj555 committed:
Update README.md

Dineshraj555 committed:
Updated java docs, improved gui and minor patch up works.

Dineshraj555 committed:
Update README.md

Dineshraj555 committed:
Merge branch 'master' of https://github.com/Dineshraj555/ip

* 'master' of https://github.com/Dineshraj555/ip:
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md
  Update README.md

Dineshraj555 committed:
Fixed storage file and directory path issues.

Dineshraj555 committed:
Final fix for exception handling.
